src/main_app/components/SignUp/index.js

```diff
@@ -0,0 +1,65 @@
+import { Formik, Field, Form } from 'formik'
+import * as Yup from 'yup'
+
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms))
+
+const SignupSchema = Yup.object().shape({
+    firstName: Yup.string()
+      .min(4, 'Too Short!')
+      .max(50, 'Too Long!')
+      .required('Required'),
+    lastName: Yup.string()
+      .min(2, 'Too Short!')
+      .max(50, 'Too Long!')
+      .required('Required'),
+    email: Yup.string().email('Invalid email').required('Required')
+})
+
+const SignUp = () => (
+    <div>
+        <h1>Sign Up</h1>
+        <Formik
+        initialValues={{
+            firstName: '',
+            lastName: '',
+            email: '',
+            password: '',
+        }}
+        validationSchema={SignupSchema}
+        onSubmit={async (values, actions) => {
+            await sleep(500);
+            alert(JSON.stringify(values, null, 2));
+            actions.setSubmitting(false);
+        }}
+        >
+        {({ isSubmitting, errors, touched }) => (
+            <Form>
+
+            <label htmlFor="firstName">First Name</label>
+            <Field name="firstName" />
+            {errors.firstName && touched.firstName ? (
+                <div>{errors.firstName}</div>
+            ) : null}<br></br>
```

Comment:
- Formik provides a component to show error messages. Take a look at [`ErrorMessage`](https://formik.org/docs/api/errormessage)
- Don't use `<br />` to create a space between elements. Use a `margin` with CSS.